File: python/triangle/triangle.py
def equilateral(sides):
    return triangle(sides) and sides[0] == sides[1] and sides[1] == sides[2]


def isosceles(sides):
    return triangle(sides) and (sides[0] == sides[1] or sides[1] == sides[2] or sides[0] == sides[2])


def scalene(sides):
    return triangle(sides) and sides[0] != sides[1] and sides[1] != sides[2] and sides[0] != sides[2]


def triangle(sides):
    return sides[0]+sides[1] > sides[2] and sides[1]+sides[2] > sides[0] and sides[0]+sides[2] > sides[1]

File: solutions/ruby/triangle/2/triangle.rb
=begin
Write your code for the 'Triangle' exercise in this file. Make the tests in
`triangle_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/triangle` directory.
=end

class Triangle
  def initialize(sizes)
    @sizes = sizes
  end

  def sizes
    @sizes
  end

  def triangle?
    sizes[0] + sizes[1] > sizes[2] && sizes[1] + sizes[2] > sizes[0] && sizes[0] + sizes[2] > sizes[1]
  end

  def equilateral?
    triangle? && sizes[0] == sizes[1] && sizes[1] == sizes[2]
  end

  def isosceles?
    triangle? && (sizes[0] == sizes[1] || sizes[1] == sizes[2] || sizes[0] == sizes[2])
  end

  def scalene?
    triangle? && sizes[0] != sizes[1] && sizes[1] != sizes[2] && sizes[0] != sizes[2]
  end
end
